src/libasr/asr_verify.cpp

```diff
@@ -578,21 +578,23 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
             require(x_m_module_name == asr_owner_name,
                 "ExternalSymbol::m_module_name `" + x_m_module_name
                 + "` must match external's module name `" + asr_owner_name + "`");
-            ASR::symbol_t *s = nullptr;
-            if( m != nullptr && ((ASR::symbol_t*) m == ASRUtils::get_asr_owner(x.m_external)) ) {
-                s = m->m_symtab->find_scoped_symbol(x.m_original_name, x.n_scope_names, x.m_scope_names);
-            } else if( sm ) {
-                s = sm->m_symtab->resolve_symbol(std::string(x.m_original_name));
-            } else if( em ) {
-                s = em->m_symtab->resolve_symbol(std::string(x.m_original_name));
+            if (asr_owner_name != "_global_symbols") {
```

Comment:
Why is special handling required here?

I would document it why it is needed. I think we should check `_global_symbols` too.

Reply:
The issue for which I made these changes, seems to be fixed. So, I'm reverting this back.
Sorry, I didn't notice this before.

Reply:
ok!